token/cli/src/main.rs

```diff
@@ -1060,7 +1132,12 @@ async fn command_transfer(
         Some(mint_info.decimals)
     };
 
-    let token = token_client_from_config(config, &token_pubkey, decimals)?;
+    let token = if let Some(transfer_hook_accounts) = transfer_hook_accounts {
+        token_client_from_config(config, &token_pubkey, decimals)?
+            .with_transfer_hook_accounts(transfer_hook_accounts)
+    } else {
+        token_client_from_config(config, &token_pubkey, decimals)?
+    };
```

Comment:
It's mostly for offline signing, since the validation account is required for the interface. maybe that means we can just include it if any transfer hook accounts are included :thinking:

Reply:
Ahh right because transfer hook program and validation account are expected in this vector, ie. https://github.com/solana-labs/solana-program-library/blob/37d21c6493de474f72e247931dbb42274ea002ba/token/transfer-hook-interface/src/offchain.rs#L57C5-L61C78

Because of the way that offchain helper is written, you'd have to validate that they haven't included either already and add them for them, or don't if they have. But might be a nice touch for DX :)

Reply:
Idk, I am leaning towards it might not be necessary to try to automagically include any metas for them for offline signing. Keep it simple. If you want to sign offline add each key according to: https://github.com/solana-labs/solana-program-library/blob/master/token/transfer-hook-interface/src/instruction.rs#L22C5-L27C82

Reply:
Actually, we don't even know the program id when we're offline, so we can't even derive the proper address, unless we keep the transfer hook program id as a separate argument. Yeah, let's keep it simple